source/-XML/XMLWriter.cpp

```diff
@@ -386,7 +387,8 @@ writeRecord(
 
     writeSymbol(I);
 
-    writeTemplate(I.Template);
+    if(I.Template)
+        writeTemplate(*I.Template);
```

Comment:
consider if you want to do the emptiness check inside `writeTemplate`. For example:
```
void writeTemplate( std::unique_ptr<TemplateInfo> const& );
```

This makes the call site look nice and tidy

Reply:
This is going to be changed once implicit specializations are implemented fully, so I'd like to just leave it as-is for now